tests/integration/scenarios.py

```diff
@@ -134,6 +134,48 @@ def test_remove_virtual_interfaces(self):
         res = subprocess.run(['ip', 'link', 'show', 'dev', 'br54'], capture_output=True, text=True)
         self.assertIn('not exist', res.stderr)
 
+    # https://bugs.launchpad.net/netplan/+bug/2083029
+    def test_netplan_try(self):
+        self.setup_eth(None)
+        with open(self.config, 'w') as f:
+            f.write('''network:
+  renderer: %(r)s
+  ethernets:
+    %(e2c)s:
+      match: {name: %(e2c)s}
+      dhcp4: yes''' % {'r': self.backend, 'e2c': self.dev_e2_client})
+        self.generate_and_settle([self.state_dhcp4(self.dev_e2_client)])
+        self.assert_iface_up(self.dev_e2_client,
+                             ['inet 192.168.6.[0-9]+/24'])
+
+        with open(self.config, 'w') as f:
+            f.write('''network:
+  renderer: %(r)s
+  ethernets:
+    %(e2c)s:
+      match: {name: %(e2c)s}
+      dhcp4: false
+      addresses: [192.168.6.201/24]''' % {'r': self.backend, 'e2c': self.dev_e2_client})
+
+        def assert_static_ip():
+            self.assert_iface_up(self.dev_e2_client,
+                                 ['inet 192.168.6.201/24'])
+
+        self.try_and_settle(assert_static_ip,
+                            wait_interfaces_try=[
+                                self.state(self.dev_e2_client, "inet 192.168.6.201"),
+                            ],
+                            wait_interfaces_after=[
+                                # Use "dynamic" instead of state_dhcp4 since
+                                # state_dhcp4 will match our static address.
+                                self.state(self.dev_e2_client, "dynamic"),
```

Comment:
suggestion (non-blocking): Should we use another IP range, e.g. `10.1.2.3/24` to avoid confusion with the DHCP4 range for eth42 (`192.168.6.10..192.168.6.200`)?

Reply:
Wonder about defining a static lease for the test interfaces instead, that way `state_dhcp4` can just match the IP from the static lease

Reply:
Yeah, I like the idea, but I'm not sure if it would be working for all the tests. Sometimes we add additional interfaces like bridges and try to get a DHCP lease on those, potentially multiple interfaces. So a single, static lease might not be enough.

Anyway, I considered this non-blocking. So let's keep it as-is for now!